BackendTaskPilot/Program.cs

```diff
@@ -0,0 +1,38 @@
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+ExchangeRateService service = new ExchangeRateService();
+
+// REST API to get exchange rates based on input currencies
+app.MapPost("/", service.GetExchangeRates);
+
+app.Run();
+
+public struct CnbRatesResponse
+{
+    [JsonPropertyName("rates")] public List<Rate> Rates { get; set; }
+}
+
+public struct Rate
+{
+    [JsonPropertyName("currencyCode")] public string CurrencyCode { get; set; }
+    [JsonPropertyName("rate")] public float RateValue { get; set; }
+}
+
+internal class ExchangeRateService
```

Comment:
The ExchangeRateService class can be improved by following SOLID principles
- Implement the IExchangeRateService interface
- Place IExchangeRateService interface and ExchangeRateService in separate files with same name respectively
- Move both into the `ExchangeRate.Application` project to separate the business code from http api code

Use the factory design pattern to allow the code be easily extended to other source currencies in the future. 
Implement two interfaces similar to below

Each source currency will implement this interface to retrieve the rates from the official external api
```csharp
public interface IExchangeRateProvider
{
    public string CurrencyCode { get; init; }
    Task<IEnumerable<ExchangeRate>> GetDailyExchangeRatesAsync(DateOnly day, CancellationToken cancellationToken);
}
```

// This interface has a contract to return the provider for a given source currency
```csharp
public interface IExchangeRateProviderFactory
{
    IExchangeRateProvider Get(string currencyCode);
}
```